primitives/core/src/crypto.rs

```diff
@@ -291,7 +287,7 @@ pub trait Ss58Codec: Sized + AsMut<[u8]> + AsRef<[u8]> + Default {
 	fn from_string(s: &str) -> Result<Self, PublicError> {
 		Self::from_string_with_version(s).and_then(|(r, v)| match v {
 			v if !v.is_custom() => Ok(r),
-			v if v == *DEFAULT_VERSION.lock() => Ok(r),
+			v if v == ss58_address_format() => Ok(r),
```

Comment:
having it named ss58_address_format makes it a bit difficult to understand. Like for me Ss58AddressFormat is a type, not a specific value of the type.
Maybe we can rename `default_ss58_version` ?
and `unwrap_or_default_ss58_version`.
So it is consistent with `set_default_ss58_version`.